cdragontoolbox/wad.py

```diff
@@ -222,6 +222,18 @@ def set_unknown_paths(self, path):
                 else:
                     wadfile.path = f"{path}/{wadfile.path_hash:016x}"
 
+    def sanitize_paths(self):
+        """Truncate files whose basename has a length of at least 255"""
+
+        for wadfile in self.files:
+            if wadfile.path:
+                path, filename = os.path.split(wadfile.path)
+                if len(filename) < 255:
+                    continue
+
+                basename, ext = os.path.splitext(filename)
+                wadfile.path = os.path.join(path, f"{basename[:255-21]}.{hex(wadfile.path_hash)[2:]}{ext}")
```

Comment:
- I didn't realized during the first path, but the lenght from the original `basename` should depend on `ext`' lengths.
- Don't use `{hex(...)}`, use `{value:016x}` to format `value` as hex (16 digits, with leading 0s).

Reply:
> but the lenght from the original basename should depend on ext' lengths

That's true, but I'm pretty much assuming I will always have a bin file here, so I thought it was fine as-is.

Reply:
It's usually better to adopt a generic solution from the start when it does not cost much more (as here).

Reply:
Okay, should be resolved now.